llvm/lib/Analysis/BasicAliasAnalysis.cpp

```diff
@@ -268,43 +268,59 @@ struct CastedValue {
   unsigned ZExtBits = 0;
   unsigned SExtBits = 0;
   unsigned TruncBits = 0;
+  bool IsNonNegative = false;
```

Comment:
`/// Whether trunc(V) is non-negative.` Should clarify that this does not apply to `V`.